test/e2e/specs/editor/plugins/nonce.spec.js

```diff
@@ -0,0 +1,31 @@
+/**
+ * WordPress dependencies
+ */
+const { test, expect } = require( '@wordpress/e2e-test-utils-playwright' );
+
+test.describe( 'Nonce', () => {
+	// While using beforeEach/afterEach is suboptimal for multiple tests, they
+	// are used here to ensure that the nonce plugin doesn't interfere with API
+	// calls made in global before/after calls, which may perform API requests.
+	test.beforeEach( async ( { requestUtils } ) => {
+		await requestUtils.activatePlugin( 'gutenberg-test-plugin-nonce' );
+	} );
+	test.afterEach( async ( { requestUtils } ) => {
+		await requestUtils.deactivatePlugin( 'gutenberg-test-plugin-nonce' );
+	} );
+
+	test( 'should refresh when expired', async ( { page, admin } ) => {
+		await admin.createNewPost();
+		await page.keyboard.press( 'Enter' );
+		// eslint-disable-next-line no-restricted-syntax
+		await page.waitForTimeout( 5000 );
+		await page.keyboard.type( 'test' );
+		// `saveDraft` waits for saving to be successful, so this test would
+		// timeout if it's not.
+		await page.click( 'role=button[name=/Save draft/i]' );
+		// We expect a 403 status once.
+		page.on( 'console', ( message ) => {
+			expect( message.type() ).toBe( 'error' );
+		} );
```

Comment:
I wonder if we could use the official [`response`](https://playwright.dev/docs/api/class-request) event for this assertion. Something like:

```js
const saveDraftResponse = await new Promise(resolve => {
  page.on('response', response => {
    if (response.url() === 'the errored url') {
      resolve(response);
    }
  });
});

expect(saveDraftResponse.status()).toBe(403);
```

Reply:
Hi @kevin940726,

I tried the above suggestion but somehow the API's response is 200.

![image](https://user-images.githubusercontent.com/41000648/189888432-140196b2-3a94-428c-ae69-47356469606b.png)

Because of which kept the console message as is..

Reply:
Hi @kevin940726,
Any further suggestions here?

Reply:
The original test here is a bit weird as it doesn't tell which response should error. My suggestion would be to try to figure out which response should error and explicitly test it in this test, then follow the code I wrote above.

Reply:
This is tricky. This test depends on the old Jest-Puppeteer behavior to auto re-auth manually, it however is not what the new testing architecture in Playwright does now though. I have a working local refactor that uses `page.route` to mock the API so that we don't need the nonce test plugin and the 5 seconds timeout. I'll share more soon.

Reply:
Sure!

Reply:
Opened https://github.com/WordPress/gutenberg/pull/44929. It's a bigger change than I thought. I figured opening a new PR might be easier in this case 😅 . Your contribution is still very helpful though! We can co-author it when it's merged ;).

Reply:
Sure, I am going through the PR you just created. 

Should I close this PR then?

Reply:
Yes, that'd be great :) Thank you!

Reply:
Sure.